tslib.es6.js

```diff
@@ -348,6 +348,18 @@ export function __disposeResources(env) {
     return next();
 }
 
+export function __rewriteRelativeImportExtension(path, preserveJsx) {
+    if (typeof path === "string" && path[0] === "." && (path[1] === "/" || path[1] === "." && path[2] === "/")) {
+        if (path.substring(path.length - 4).toLowerCase() === ".tsx") {
+            return path.substring(0, path.length - 4) + (preserveJsx ? ".jsx" : ".js");
+        }
+        return path.replace(/(\.d)?(\.[^./]+?)?\.([cm])?ts$/i, function (m, d, ext, cm) {

```

Comment:
Is there a non-negated repeated character class I can use to capture other file extensions you might want to write a declaration file for? `\.[^./]+` doesn’t account for URL-ish endings like query parameters. I don’t know what characters comprise reasonable file extensions. Is `[a-z0-9_-]` sufficient? Too permissive?

Reply:
If you're concerned about query parameters, then `[^.?#\\/]` might be sufficient. `#` is allowed in windows paths, but we could just error on a path containing that character at compile time.

`[a-z0-9_-]` (or just `[\w-]`, [since that's equivalent](https://tc39.es/ecma262/#ASCII-word-characters) in `i` mode) might be sufficient, but I'm not sure what the frequency of non-ASCII characters in file extensions might be. At worst we should match whatever we actually allow for `.d.foo.ts` in TypeScript, and as far as I can tell we just look for `.d.` in `getDeclarationFileExtension`.

Reply:
`getDeclarationFileExtension` maybe should be updated to handle `?` and `#` 🤔

Reply:
Actually, no changes are needed to the regex, because this is already capturing a case where we _don’t_ want to rewrite. Pushing a passing test and then I’m satisfied.